server/middleware/src/spaceSecurity.ts

```diff
@@ -95,6 +95,11 @@ export class SpaceSecurityMiddleware extends BaseMiddleware implements Middlewar
   }
 
   private async init (ctx: MeasureContext): Promise<void> {
+    // FIXME: The following line looks up for private spaces in DOMAIN_SPACE only.
+    //  Thus, if you have a model class that represents private spaces,
+    //  and try to store it in its own domain, e.g. `@Model(MyModel, Space, MY_DOMAIN)`,
```

Comment:
Nope, we disallow to change domain for derived classes, so we only need to extend dsl to throw an error if such case is detected.

Reply:
Might be a good idea to implement this check here, and get rid of this comment then.

> we disallow to change domain for derived classes

Is it meant only for Space and its descendants? Or any class that has already declared its domain?

Reply:
if the domain has already been defined in a class, descendant classes cannot redeclare it.

Reply:
Implemented check (which already caught non-critical violation in `model-activity`), and dropped this comment